### PR TITLE
Backwards Compatibility Fixes

### PR DESCRIPTION
**Pull request by donn**
## Steps

* `Odb.CustomIOPlacement`
  * `QUIT_ON_UNMATCHED_IO` now migrates to new variable `ERRORS_ON_UNMATCHED_IO`, an enumeration of four variables that controls whether errors are emitted in:
    * no situation
    * situations where pins in the design are missing from the config file (default, matching openlane 1)
    * situations where pins in the config file are missing from the design (new)
    * either situation

* `OpenROAD.STAPrePnR`, `OpenROAD.STAPostPnR`
  * Added new configuration variable `EXTRA_SPEFS` ONLY for backwards compatibility with OpenLane 1 that should not otherwise be used
  * Internally reworked how timing-related views are passed from Python to Tcl

## Misc. Enhancements/Bugfixes

* `openlane.common.toolbox`
  * New method, `get_timing_files_categorized`, returns the three design formats in, get this, three separate lists
 
* `openlane.config`
  * Config no longer attempts to migrate `EXTRA_SPEFS` to `MACROS` because of side-effects (e.g. the dummy paths)

## Testing
* Add an OpenLane 1-compatible configuration for `aes_user_project_wrapper` to test back-compat
* Made sure `open_proj_timer` config matches OpenLane 1

---
Depends on https://github.com/efabless/openlane2-ci-designs/pull/25
Depends on https://github.com/efabless/openlane2-step-unit-tests/pull/27

### CODE REVIEW

**Comment by kareefardi** [comment]
Can't figure out how to read in multiple GDS files for a macro in this https://github.com/efabless/openlane2/blob/b9e91d99c2fbc9f3b2b71fde5005c0f6ab03a714/openlane/scripts/magic/common/read.tcl#L77. We also never had a use case for that.

**Review by kareefardi — CHANGES_REQUESTED** [review]
> * `Odb.CustomIOPlacement`
  * `QUIT_ON_UNMATCHED_IO` now `False` by default to match OpenLane 1

OpenLane 1 default is True https://github.com/The-OpenROAD-Project/OpenLane/blob/2719508edbb579ab7317d1e0942b774d8d4aa214/configuration/floorplan.tcl#L39
This variable should be change to `ERROR_ON*`. I missed it in the earlier PR.

File: openlane/common/toolbox.py [hunk 1]
```diff
@@ -202,12 +202,12 @@ def get_macro_views_by_priority(
...
         return result
 
-    def get_timing_files(
+    def get_timing_files_categorized(
         self,
         config: Mapping[str, Any],
         timing_corner: Optional[str] = None,
         prioritize_nl: bool = False,
-    ) -> Tuple[str, List[str]]:
+    ) -> Tuple[str, List[Path], List[Path], List[Tuple[str, Path]]]:
```
Comment (kareefardi) [inline openlane/common/toolbox.py:210]: `List[Path]` repeated

File: openlane/common/toolbox.py [hunk 2]
```diff
@@ -202,12 +202,12 @@ def get_macro_views_by_priority(
             formats_so_far.append(format)
         return result
 
-    def get_timing_files(
+    def get_timing_files_categorized(
```
Comment (kareefardi) [inline openlane/common/toolbox.py:205]: that's an API break

**Review by kareefardi — CHANGES_REQUESTED** [review]
`warn_once` is a bit unorthodox. Also, It is part of addressing warnings in OpenLane and beyond the scope of this PR. It would be best if it is moved to a different PR.

**Comment by donn** [comment]
@kareefardi Fair enough. Will revert for now-- we do need a solution for repeated warnings though because it's getting a bit out of hand

File: openlane/common/toolbox.py (hunk 1 above)
Comment (donn) [reply openlane/common/toolbox.py:210]: That's intentional: corner, list of libs, list of netlists, list of instances+spefs

Comment (kareefardi) [reply openlane/common/toolbox.py:210]: Got it. Confused `Tuple` with `Union`

**Review by kareefardi — APPROVED** [review]

**Pull request merged**